server/svix-server/src/queue/redis.rs

```diff
@@ -36,89 +48,222 @@ use super::{
+                    .await
+                    .unwrap();
+                let tasks: Vec<&str> = keys
+                    .iter()
+                    // All information is stored in the key in which the ID and JSON formated task
+                    // are separated by a `|`. So, take the key, then take the part after the `|`
+                    .map(|x| &x.0)
+                    .map(|x| x.split('|').nth(1).expect("Improper key format"))
                     .collect();
-                let _: () = pool.rpush(MAIN, keys).await.unwrap();
+
+                // Then for each task, XADD them to ghe MAIN queue
+                for task in tasks {
+                    let _: () = pool
+                        .query_async(Cmd::xadd(
+                            &main_queue_name,
+                            GENERATE_STREAM_ID,
+                            &[(QUEUE_KV_KEY, task)],
+                        ))
+                        .await
+                        .unwrap();
+                }
             } else {
                 // Wait for half a second before attempting to fetch again if nothing was found
                 sleep(Duration::from_millis(500)).await;
             }
 
-            // Every iteration here also check whether the processing queue has items that should
-            // be picked back up
-            let keys: Vec<String> = pool.lrange(PROCESSING, 0isize, 1isize).await.unwrap();
-
-            // If the key is older than now, it means we should be processing keys
-            let validity_limit =
-                KsuidMs::new(Some(Utc::now() - task_validity_duration), None).to_string();
-
-            if !keys.is_empty() && keys[0] <= validity_limit {
-                let keys: Vec<String> = pool.lrange(PROCESSING, 0isize, batch_size).await.unwrap();
-                for key in keys {
-                    if key <= validity_limit {
-                        // We use LREM to be sure we only delete the keys we should be deleting
-                        tracing::trace!("Pushing back overdue task to queue {}", key);
-                        let refreshed_key = regenerate_key(&key);
-                        let _: () = pool.rpush(MAIN, &refreshed_key).await.unwrap();
-                        let _: () = pool.lrem(PROCESSING, 1, &key).await.unwrap();
-                    }
+            // Every iteration here also check whether the processing queue has items that
+            // should be picked back up
+            let mut cmd = redis::cmd("XPENDING");
+            let _ = cmd
+                .arg(&main_queue_name)
+                .arg(WORKERS_GROUP)
+                // Search only for IDs that have been idle for at least the pending_duration
+                .arg("IDLE")
+                .arg(pending_duration)
+                // And search for at most 1000 IDs from the minimum ID value to the maximum ID value
+                .arg("-")
+                .arg("+")
+                .arg(1000i16);
```

Comment:
Make this a const. In general when you have values that affect behavior, let's have them somewhere more visible so that we know more easily what we can modify to affect performance and behavior.